orbs/circleci/circle-wait-job

```diff
@@ -0,0 +1,180 @@
+    must_not_empty("current_job", @current_job)
+    must_not_empty("organization", @organization)
+    must_not_empty("project", @project)
+
+    self
+  end
+
+  private
+
+  def must_not_empty(key, value)
+    if value.to_s.empty?
+      raise(ArgumentError, "the #{key} is missing or empty")
+    end
+  end
+
+  def define_options!(parser)
+    parser.banner = "Usage: circleci-wait-job OPTIONS"
+    parser.separator("")
+    parser.separator("OPTIONS")
+    parser.separator("")
+
+    parser.on("-t=VALUE", "--token=VALUE", "the circleci api token") { |value| @token = value }
+    parser.on("-o=VALUE", "--organization=VALUE", "the circleci organization name") { |value| @organization = value }
+    parser.on("-p=VALUE", "--project=VALUE", "the circleci project name") { |value| @project = value }
+    parser.on("-j=VALUE", "--job=VALUE", "the circleci job name") { |value| @job = value }
+    parser.on("-c=VALUE", "--current_job=VALUE", "the circleci currently running job name") { |value| @current_job = value }
+    parser.on("-b=VALUE", "--branch=VALUE", "the circleci branch") { |value| @branch = value }
+    parser.on("-i=VALUE", "--interval=VALUE", "the wait time between pooling") { |value| @interval = value }
+    parser.on("--vcs=VALUE", "the version control system (default: github)") { |value| @vcs = value }
+    parser.on("--current-build-number=VALUE", "the current circleci build number") { |value| @build_number = value }
+
+    parser.on_tail("-h", "--help", "print help and exit") do
+      puts(parser)
+      exit(0)
+    end
+
+    parser.on_tail("--version", "print version and exit") do
+      puts("v0.0.1")
+      exit(0)
+    end
+  end
+end
+
+cli = CLI.new
+
+begin
+  options = cli.parse(ARGV)
+rescue ArgumentError => e
+  puts(e.message)
+  exit(1)
+end
+
+uri = URI::HTTPS.build(host: "circleci.com",
+                       path: File.join("/",
+                                       "api",
+                                       "v1.1",
+                                       "project",
+                                       options.vcs.to_s,
+                                       options.organization.to_s,
+                                       options.project.to_s,
+                                       "tree",
+                                       options.branch.to_s),
+                       query: URI.encode_www_form({"circle-token" => options.token.to_s,
+                                                   "shallow" => true,
+                                                   "filter" => "running"}))
+
+
+def other_job_is_running?(uri, job, current_job, build_num)
+  response = Net::HTTP.get(uri)
+  builds = JSON.parse(response)
+
+  # Check if another identical job (most likely a production deployment job) is running for another build
+  has_concurrent_job = builds.any? do |build|
+    is_concurrent_build = build.dig("workflows", "job_name").to_s == job && build["build_num"] != build_num
```

Comment:
Since the circle-wait-job is run in a dedicated build, we don't need to check anymore build number.
```suggestion
    is_concurrent_build = build.dig("workflows", "job_name").to_s == job
```

Reply:
Thanks, it's fixed!